rclcpp/src/rclcpp/executor.cpp

```diff
@@ -266,7 +266,7 @@ Executor::execute_subscription(
   if (ret == RCL_RET_OK) {
     message_info.from_intra_process = false;
     subscription->handle_message(message, message_info);
-  } else if (ret != RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
+  } else if (ret == RCL_RET_SUBSCRIPTION_TAKE_FAILED) {
```

Comment:
Hmm, ok. I am confused by the comparison though.
It says "if there is no subscription take failure, then print error 'take failed ..'